runtime/gc_vlhgc/RememberedSetCardBucket.cpp

```diff
@@ -213,13 +217,16 @@ MM_RememberedSetCardBucket::compact(MM_EnvironmentVLHGC *env)
 
 			/* find top index for this buffer */
 			UDATA fromIndexTop = MAX_BUFFER_SIZE;
-			if (isCurrentSlotWithinBuffer(fromBufferCardList)) {
-				fromIndexTop = _current -  fromBufferCardList;
+			if (isCurrentSlotWithinBuffer(env, fromBufferCardList)) {
+				fromIndexTop = MM_RememberedSetCard::subtractCardAddresses(_current, fromBufferCardList, compressed);
 			}
 
 			for (UDATA fromIndex = 0; fromIndex < fromIndexTop; fromIndex++) {
-				if (0 != fromBufferCardList[fromIndex]) {
-					toBufferCardList[toIndex++] = fromBufferCardList[fromIndex];
+				MM_RememberedSetCard *fromAddress = MM_RememberedSetCard::addToCardAddress(fromBufferCardList, fromIndex, compressed);
+				UDATA card = MM_RememberedSetCard::readCard(fromAddress, compressed);
+				if (0 != card) {
+					MM_RememberedSetCard *toAddress = MM_RememberedSetCard::addToCardAddress(toBufferCardList, toIndex++, compressed);
+					MM_RememberedSetCard::writeCard(toAddress, card, compressed);
```

Comment:
Would you please remove `++` from `toIndex++` but add `toIndex += 1;` after write?